frontend/app/src/components/base/BaseSecure.vue

```diff
@@ -0,0 +1,51 @@
+<script>
+import { mapActions, mapState } from 'pinia';
+import { useAuthStore } from '~/store/auth';
+
+export default {
+  props: {
+    rolesCanAccess: {
+      type: Array,
+      default: undefined,
+    },
+  },
+  computed: {
+    ...mapState(useAuthStore, [
+      'authenticated',
+      'userCurrentRoles',
+      'isAdmin',
+      'isUser',
+      'ready',
+    ]),
+  },
+  methods: mapActions(useAuthStore, ['login']),
+};
+</script>
+
+<template>
+  <div v-if="authenticated">
+    <div v-if="userCurrentRoles.some((role) => rolesCanAccess.includes(role))">
+      <slot />
+    </div>
+    <!-- TODO: Figure out better way to alert when user lacks chefs user role -->
+    <div v-else class="text-center">
+      <h1 class="my-8">401 UnAuthorized</h1>
+      <p>
+        <span>Your account is not set up correctly.</span>
+      </p>
+    </div>
+  </div>
+  <div v-else class="text-center">
+    <h1 class="my-8">Please login to your account</h1>
```

Comment:
As we don't have any Public facing page so yes we need to redirect user to Login page. Initially there was a welcome page that could be rendered without the authorization but for some reason team wants us to remove that page and make the File Task Management is Home.

Reply:
ok, what I was trying to say is that instead of giving a button to login, can't we take him to login page of keycloak. If that is not the requirement, then please ignore.